cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ETLMapReduce.java

```diff
@@ -356,17 +356,12 @@ public void act(DefaultJoinerContext joinerContext) {
     hConf.set(INPUT_ALIAS_KEY, GSON.toJson(inputAliasToStage));
     finisher = new CompositeFinisher(finishers);
 
-    // setup time partition for each error dataset
-    for (StageSpec stageInfo : Sets.union(phase.getStagesOfType(Transform.PLUGIN_TYPE),
-                                          phase.getStagesOfType(BatchSink.PLUGIN_TYPE))) {
-      if (stageInfo.getErrorDatasetName() != null) {
-        Map<String, String> args = new HashMap<>();
-        args.put(FileSetProperties.OUTPUT_PROPERTIES_PREFIX + "avro.schema.output.key",
-                 Constants.ERROR_SCHEMA.toString());
-        TimePartitionedFileSetArguments.setOutputPartitionTime(args, context.getLogicalStartTime());
-        context.addOutput(Output.ofDataset(stageInfo.getErrorDatasetName(), args));
+    context.execute(new TxRunnable() {
```

Comment:
if we don't want this to be required, should put a comment here with a link to a relevant jira.

Reply:
Created a (JIRA)[https://issues.cask.co/browse/CDAP-12634] and updated this line to be preceded with a comment.